### PR TITLE
Add PlanBuilder::partitionedOutputArbitrary

### PR DESCRIPTION
**Pull request by kewang1024**
Create a PartitionedOutputNode to output the input data in arbitrary buffer.

### CODE REVIEW

File: velox/exec/tests/utils/PlanBuilder.h [hunk 1]
```diff
@@ -569,6 +569,10 @@ class PlanBuilder {
   PlanBuilder& partitionedOutputBroadcast(
       const std::vector<std::string>& outputLayout = {});
 
+  /// Add a PartitionedOutputNode to output the input data in arbitrary buffer.
```
Comment (mbasmanova) [inline velox/exec/tests/utils/PlanBuilder.h:572]: typos: "to output the input data in"

to put data into arbitrary buffer

**Review by mbasmanova — APPROVED** [review]
Thanks.

File: velox/exec/tests/utils/PlanBuilder.h (hunk 1 above)
Comment (xiaoxmeng) [inline velox/exec/tests/utils/PlanBuilder.h:572]:
```
nit: s/Add/Adds/
```

**Review by xiaoxmeng — APPROVED** [review]
@kewang1024 LGTM. Thanks!

**Pull request closed without merging**